src/dbms/replication_handler.cpp

```diff
@@ -214,20 +226,26 @@ bool ReplicationHandler::IsReplica() const { return dbms_handler_.ReplicationSta
 
 // Per storage
 // NOTE Storage will connect to all replicas. Future work might change this
-void RestoreReplication(replication::ReplicationState &repl_state, storage::Storage &storage) {
+void RestoreReplication(replication::ReplicationState &repl_state, DatabaseAccess db_acc) {
   spdlog::info("Restoring replication role.");
 
   /// MAIN
-  auto const recover_main = [&storage](RoleMainData &mainData) {
+  auto const recover_main = [db_acc = std::move(db_acc)](RoleMainData &mainData) mutable {  // NOLINT
     // Each individual client has already been restored and started. Here we just go through each database and start its
     // client
     for (auto &instance_client : mainData.registered_replicas_) {
-      spdlog::info("Replica {} restoration started for {}.", instance_client.name_, storage.id());
+      spdlog::info("Replica {} restoration started for {}.", instance_client.name_, db_acc->name());
 
-      const auto &ret = storage.repl_storage_state_.replication_clients_.WithLock(
-          [&](auto &storage_clients) -> utils::BasicResult<RegisterReplicaError> {
+      // Phase 1: ensure DB exists
+      // auto s = instance_client.rpc_client_.Stream<memgraph::replication::????>();
```

Comment:
I'm unsure about the state machine at this point. I feel is client is system behind than that implies db is maybe behind

Reply:
Not sure.
We could set all database replication state to behind. Perhaps not needed if the replica has recovery and no delta has been sent in the meantime.
However the show replicas state is misleading.

Reply:
Lets deliver this as part of the future `SHOW REPLICAS` PR.